phpunit/class-rest-blocks-controller-test.php

```diff
@@ -1,341 +0,0 @@
-		$this->assertEquals( 200, $response->get_status() );
-
-		$data = $response->get_data();
-
-		$this->assertArrayHasKey( 'id', $data );
-		$this->assertArrayHasKey( 'title', $data );
-		$this->assertArrayHasKey( 'content', $data );
-
-		$this->assertEquals( self::$post_id, $data['id'] );
-		$this->assertEquals( 'Updated cool block', $data['title'] );
-		$this->assertEquals( '<!-- wp:core/paragraph --><p>Nice!</p><!-- /wp:core/paragraph -->', $data['content'] );
-	}
-
-	/**
-	 * Check that we can DELETE a block.
-	 */
-	public function test_delete_item() {
-		wp_set_current_user( self::$user_id );
-
-		$request = new WP_REST_Request( 'DELETE', '/wp/v2/blocks/' . self::$post_id );
-
-		$response = rest_get_server()->dispatch( $request );
-
-		$this->assertEquals( 200, $response->get_status() );
-
-		$data = $response->get_data();
-
-		$this->assertArrayHasKey( 'deleted', $data );
-		$this->assertArrayHasKey( 'previous', $data );
-
-		$this->assertTrue( $data['deleted'] );
-
-		$this->assertArrayHasKey( 'id', $data['previous'] );
-		$this->assertArrayHasKey( 'title', $data['previous'] );
-		$this->assertArrayHasKey( 'content', $data['previous'] );
-
-		$this->assertEquals( self::$post_id, $data['previous']['id'] );
-		$this->assertEquals( 'My cool block', $data['previous']['title'] );
-		$this->assertEquals( '<!-- wp:core/paragraph --><p>Hello!</p><!-- /wp:core/paragraph -->', $data['previous']['content'] );
-	}
-
-	/**
-	 * Check that we have defined a JSON schema.
-	 */
-	public function test_get_item_schema() {
-		$request    = new WP_REST_Request( 'OPTIONS', '/wp/v2/blocks' );
-		$response   = rest_get_server()->dispatch( $request );
-		$data       = $response->get_data();
-		$properties = $data['schema']['properties'];
-
-		$this->assertEquals( 3, count( $properties ) );
-		$this->assertArrayHasKey( 'id', $properties );
-		$this->assertArrayHasKey( 'title', $properties );
-		$this->assertArrayHasKey( 'content', $properties );
-	}
-
-	/**
-	 * Test cases for test_capabilities().
-	 */
-	public function data_capabilities() {
-		return array(
-			array( 'create', 'editor', 201 ),
-			array( 'create', 'author', 201 ),
-			array( 'create', 'contributor', 403 ),
-			array( 'create', null, 401 ),
-
-			array( 'read', 'editor', 200 ),
-			array( 'read', 'author', 200 ),
-			array( 'read', 'contributor', 200 ),
-			array( 'read', null, 401 ),
-
-			array( 'update_delete_own', 'editor', 200 ),
-			array( 'update_delete_own', 'author', 200 ),
-			array( 'update_delete_own', 'contributor', 403 ),
-
-			array( 'update_delete_others', 'editor', 200 ),
-			array( 'update_delete_others', 'author', 403 ),
-			array( 'update_delete_others', 'contributor', 403 ),
-			array( 'update_delete_others', null, 401 ),
-		);
-	}
-
-	/**
-	 * Exhaustively check that each role either can or cannot create, edit,
-	 * update, and delete reusable blocks.
-	 *
-	 * @dataProvider data_capabilities
-	 */
-	public function test_capabilities( $action, $role, $expected_status ) {
```

Comment:
>I'd like to keep this integration test.

Restored in e3b127985e219837bab14ee0ab8a2b2e2c0aaf0b.

For what it's worth, it did prove to be a very useful test! I had found we in-fact need the controller for `check_read_permission`, because the default REST post controller otherwise ignores capabilities if the post is published.

https://github.com/WordPress/WordPress/blob/da7a80d67fea29c2badfc538bfc01c8a585f0cbe/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L1319

Discussed in Slack:

https://wordpress.slack.com/archives/C02RQC26G/p1539972633000100